### PR TITLE
Enable integration tests for MSHV

### PR DESCRIPTION
**Pull request by russell-islam**

### CODE REVIEW

File: tests/integration.rs [hunk 1]
```diff
@@ -2386,11 +2403,13 @@ mod tests {
         }
 
         #[test]
+        #[cfg(not(feature = "mshv"))]
         fn test_power_button() {
             _test_power_button(false);
```
Comment (rbradford) [inline tests/integration.rs:2408]: How does this one fail?

File: tests/integration.rs [hunk 2]
```diff
@@ -590,6 +590,17 @@ mod tests {
...
+            format! {"{{\"cpus\":{{\"boot_vcpus\":{},\"max_vcpus\":{}}},\"kernel\":{{\"path\":\"{}\"}},\"cmdline\":{{\"args\": \"{}\"}},\"net\":[{{\"ip\":\"{}\", \"mask\":\"255.255.255.0\", \"mac\":\"{}\"}}], \"disks\":[{{\"path\":\"{}\"}}, {{\"path\":\"{}\"}}]}}",
+                     cpu_count,
+                     cpu_count,
+                     direct_kernel_boot_path().to_str().unwrap(),
+                     DIRECT_KERNEL_BOOT_CMDLINE,
+                     self.network.host_ip,
+                     self.network.guest_mac,
+                     self.disk_config.disk(DiskType::OperatingSystem).unwrap().as_str(),
+                     self.disk_config.disk(DiskType::CloudInit).unwrap().as_str(),
+            }
```
Comment (sboeuf) [inline tests/integration.rs:603]: If the only difference is the kernel parameter, could you please factorize it behind a variable and pass this variable to the `format!()` function on line 587?

Comment (russell-islam) [reply tests/integration.rs]: Does the latest change make sense?

File: tests/integration.rs (hunk 1 above)
Comment (russell-islam) [reply tests/integration.rs]: It was a mistake. Fixed now.

**Review by sboeuf — APPROVED** [review]

File: tests/integration.rs (hunk 2 above)
Comment (sboeuf) [reply tests/integration.rs]: Yes thanks

**Review by rbradford — COMMENTED** [review]
There are still a lot of tests disabled and to my eyes they don't all have the same thing in common. Please open issues for the root cause of why those tests fail. e.g. `test_huge_memory` is not testing anything particularly unusual.

File: scripts/test-util.sh [hunk 3]
```diff
@@ -33,8 +33,8 @@ process_common_args() {
 	esac
 	shift
     done
-    if [[ "$hypervisor" != "kvm" ]]; then
-        echo "Hypervisor value must be kvm"
-        exit 1
+    if [[ ! ("$hypervisor" = "kvm" ||  "$hypervisor" = "mshv") ]]; then
+        die "Hypervisor value must be kvm or mshv"
     fi
+
```
Comment (likebreath) [inline scripts/test-util.sh:39]: nit: unnecessary newline

File: tests/integration.rs [hunk 4]
```diff
@@ -2458,6 +2466,7 @@ mod tests {
         }
 
         #[test]
+        #[cfg(not(feature = "mshv"))]
         fn test_guest_numa_nodes() {
```
Comment (likebreath) [inline tests/integration.rs:2470]: Given the basic NUMA test is not working for MSHV, what about the `test_live_migration_numa`?

File: tests/integration.rs [hunk 5]
```diff
@@ -5484,6 +5502,7 @@ mod tests {
         }
 
         #[test]
+        #[cfg(not(feature = "mshv"))]
         fn test_ovs_dpdk() {
```
Comment (likebreath) [inline tests/integration.rs:5506]: Ditto: given the basic "test_ovs_dpdk" test is not working for MSHV, what about the `test_live_migration_ovs_dpdk`?

**Review by likebreath — CHANGES_REQUESTED** [review]

**Comment by russell-islam** [comment]
> There are still a lot of tests disabled and to my eyes they don't all have the same thing in common. Please open issues for the root cause of why those tests fail. e.g. `test_huge_memory` is not testing anything particularly unusual.

Most of the test cases are disabled due to the unavailability of the features.  I am going to create issues for those.

**Review by likebreath — APPROVED** [review]

**Comment by likebreath** [comment]
The CI failure/timeout should be irrelevant, which is also observed from the main branch. 

I saw the following error messages, but not sure what causes the failure. Any thoughts? @rbradford @sboeuf 

```
[2021-09-27T22:10:49Z ERROR vhost_user_net] Error from the main thread: HandleRequest(PartialMessage)

[2021-09-27T22:10:49Z ERROR vhost_user_block] Error from the main thread: HandleRequest(PartialMessage)
```

**Comment by russell-islam** [comment]
> The CI failure/timeout should be irrelevant, which is also observed from the main branch.
> 
> I saw the following error messages, but not sure what causes the failure. Any thoughts? @rbradford @sboeuf
> 
> ```
> [2021-09-27T22:10:49Z ERROR vhost_user_net] Error from the main thread: HandleRequest(PartialMessage)
> 
> [2021-09-27T22:10:49Z ERROR vhost_user_block] Error from the main thread: HandleRequest(PartialMessage)
> ```

We have been seeing this error for months in our internal CI, even though the runs succeed with ok at the end.

**Comment by sboeuf** [comment]
You can disregard these error messages. The problem was coming from `virtiofsd-rs`, but it's been fixed now.

**Comment by sboeuf** [comment]
This PR's ready and the issues corresponding to the missing tests have been opened. @rbradford you can go ahead and merge it if there's nothing left on your side.

**Review by rbradford — APPROVED** [review]

**Comment by sdake** [comment]
@rbradford this issue has been open for a long time but looks done to me. Any chance of making an issue that tracks the bring-up on MSHV? I am not familiar with MSHV, but happy to produce an issue that helps better define what is missing from MSHV from a feature perspective in the project that may be a problem of the upstream product versus the stability and status of MSHV as part of issue #4121 .

**Comment by rbradford** [comment]
@sdake Did you mean to comment on a closed PR? Your comment looks like it was intended for an issue? There are multiple open issues about MSHV gaps so perhaps you create a tracking issue to link those together.

**Pull request merged**